src/NuGet.Services.EndToEnd/Support/Clients/V2V3SearchClient.cs

```diff
@@ -15,15 +17,26 @@ namespace NuGet.Services.EndToEnd.Support
 {
     public class V2V3SearchClient
     {
+        /// <summary>
+        /// In case cloud service information is not provided, this will be the fallback instance count.
+        /// </summary>
+        private const int DefaultServiceInstanceCount = 2;
```

Comment:
If we assume 2 and the actual count is lower, we have polling errors since some of the ports will not be listening. If we assume 2 and the actual count is higher, the `// Assert` step fail since the `// Act` step hit a search endpoint not polled.

The first case is pretty easy to identify but the second case would be very frustrating, even when testing locally.